smdebug/tensorflow/collection.py

```diff
@@ -124,11 +129,18 @@ def set_tensor_ref(self, tensor, tensor_name: tf.Tensor = None):
         else:
             name = tensor.name
             export_name = tensor.export_name
+        get_logger().debug(f"In set_tensor_ref: tensor_name:{name} export_name:{export_name}")
         self._tensors[name] = tensor
         self.add_tensor_name(export_name)
 
-    def has_tensor(self, name):
-        # tf object name
+        if name != export_name:
+            get_logger().debug(
+                f"Export_name:{export_name} != name:{name} . Adding export_name:{export_name} to include in "
+                f"collection collection_name:{self.name} selfId:{id(self)} "
+            )
+        self.include("^" + export_name + "$")
```

Comment:
This doesn't look correct. name is supposed to be the the actual name of tensor from TF's point of view. export_name is a more meaningful name for that tensor for user's benefit. Including export name is not useful and in fact might be wrong in some situations if that refers to a different tensor.

Reply:
do we use export name in case of keras or for other cases also?
keras regex match works on export_name i guess.

Reply:
export name was only used for Keras AFAIK. @NihalHarish

Reply:
Why did you need this though?

Reply:
for Tf estimator , graph is different for train and eval phase. Loss tensors are getting collected from loss collecection. 
During eval phase loss collection is empty. So we don't save any tensor. 

This change ensures that if we are saving some tensor, we add the name of the tensor being saved as regex, so that we always save these tensors if name is part of graph.

Reply:
So, question is if export_name is there, is it used for searching collection regex? 
In other words - 
in what cases tensor_name won't be there passed to this function and 
what(name or export name?) is being used for searching collection regex in these cases ?